app/initializers/irene.js

```diff
@@ -26,12 +26,9 @@ const initialize = function(application) {
   if(runtimeConfig) {
     ENV.host = runtimeConfig.IRENE_API_HOST || ENV.host;
     var devicefarmEnv = runtimeConfig.IRENE_DEVICEFARM_URL;
+    var deviceFarmPath = "websockify";
     if(devicefarmEnv) {
-      let deviceFarmWebsockifyHost = new URL(devicefarmEnv);
-      let deviceFarmSsl = deviceFarmWebsockifyHost.protocol === "wss:";
-      ENV.deviceFarmSsl = deviceFarmSsl;
-      ENV.deviceFarmPort = deviceFarmWebsockifyHost.port || (deviceFarmSsl ? 443:80);
-      ENV.deviceFarmHost = deviceFarmWebsockifyHost.hostname;
+      ENV.deviceFarmURL = `${devicefarmEnv}/${deviceFarmPath}`
```

Comment:
use new URL to validate the url

Reply:
use new URL to generate the url rather than appending. The env value may already contain `/`